core-feature-pack/src/main/resources/content/bin/domain.ps1

```diff
@@ -4,5 +4,83 @@
 #                                                                          ##
 #############################################################################
 
-# for now just call domain.bat, so we can have complete set of ps scripts
-& .\domain.bat
+$RESOLVED_JBOSS_HOME = (Get-ChildItem $MyInvocation.MyCommand.Path).Directory.Parent.FullName
+. "$RESOLVED_JBOSS_HOME\bin\common.ps1"
+
+$SERVER_OPTS = Process-Script-Parameters -Params $ARGS
+
+# Read an optional running configuration file
+$DOMAIN_CONF = $JBOSS_HOME + '\bin\domain.conf.ps1'
+$DOMAIN_CONF = Get-Env RUN_CONF $DOMAIN_CONF
+. $DOMAIN_CONF
+
+Set-Global-Variables-Domain
+
+# consolidate the host-controller and command line opts
+$HOST_CONTROLLER_OPTS=$HOST_CONTROLLER_JAVA_OPTS+$SERVER_OPTS
+# process the host-controller options
+foreach($p in $HOST_CONTROLLER_OPTS){
+	if ($p -eq $null){# odd but could happen
+		continue
+	}
+	$arg = $p.Trim()
+	if ($arg.StartsWith('-Djboss.domain.base.dir')){
+		$JBOSS_BASE_DIR=$p.Substring('-Djboss.domain.base.dir='.Length)
+	}elseif ($arg.StartsWith('-Djboss.domain.log.dir')){
+		$JBOSS_LOG_DIR=$p.Substring('-Djboss.domain.log.dir='.Length)
+	}elseif ($arg.StartsWith('-Djboss.domain.config.dir')){
+		$JBOSS_CONFIG_DIR=$p.Substring('-Djboss.domain.config.dir='.Length)
+	}
+}
+
+# If the -Djava.security.manager is found, enable the -secmgr and include a bogus security manager for JBoss Modules to replace
+# Note that HOST_CONTROLLER_JAVA_OPTS will not need to be handled here
+
+if ( $PROCESS_CONTROLLER_JAVA_OPTS -contains 'java.security.manager') {
+    echo "ERROR: The use of -Djava.security.manager has been removed. Please use the -secmgr command line argument or SECMGR=true environment variable."
+    exit
+}
+
+# Set debug settings if not already set
+if ($global:DEBUG_MODE){
+    if (-not $PROCESS_CONTROLLER_JAVA_OPTS -contains '-agentlib:jdwp'){
+        $PROCESS_CONTROLLER_JAVA_OPTS+= "-agentlib:jdwp=transport=dt_socket,address=$global:DEBUG_MODE,server=y,suspend=n"
+    }else{
+        echo "Debug already enabled in JAVA_OPTS, ignoring --debug argument"
+    }
+}
+
+Display-Environment
+
+$MODULE_OPTS += "-jvm"
+$MODULE_OPTS += "$JAVA"
+$MODULE_OPTS += "-jboss-home"
+$MODULE_OPTS += "$JBOSS_HOME"
+         
+
+$PROG_ARGS = @()
+$PROG_ARGS +='-D[Process Controller]' 
+$PROG_ARGS += $PROCESS_CONTROLLER_JAVA_OPTS
+$PROG_ARGS += "-Dorg.jboss.boot.log.file=$JBOSS_LOG_DIR/process-controller.log"
+$PROG_ARGS += "-Dlogging.configuration=file:$JBOSS_CONFIG_DIR/logging.properties"
+$PROG_ARGS += "-Djboss.home.dir=$JBOSS_HOME"
+$PROG_ARGS += "-jar"
+$PROG_ARGS += "$JBOSS_HOME\jboss-modules.jar"
```

Comment:
It probably doesn't matter, but everywhere else we use the `\` delimiter. Should it be that here too?